include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -313,6 +200,97 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
   return overallObjective;
 }
 
+//! Initialize parameters.
+template <typename SelectionPolicyType,
+          typename WeightPolicyType,
+          typename UpdatePolicyType>
+template<typename MatType>
+inline void CMAES<SelectionPolicyType, WeightPolicyType, UpdatePolicyType>::
+Initialize(MatType& iterate)
+{
+  niter = 0;
+  countval = 0;
+  eigenval = 0;
+  chi = std::sqrt(iterate.n_elem)*(1.0 - 1.0 / (4.0 * iterate.n_elem) + 
+      1.0 / (21 * std::pow(iterate.n_elem, 2)));
+  if(lambda == 0)
+    lambda = (4 + std::round(3 * std::log(iterate.n_elem))) * 10;
+
+  mu = std::round(lambda / 2);
+
+  // Generate raw weights first to compute mu_effective.
+  weightPolicy.GenerateRaw(lambda);
+  mu_eff = weightPolicy.MuEff();
+
+  // Strategy parameter setting: Adaption.
+  cc = (4.0 + mu_eff / iterate.n_elem) / 
+      (4.0 + iterate.n_elem + 2 * mu_eff / iterate.n_elem);
+  csigma = (mu_eff + 2.0) / (iterate.n_elem + mu_eff + 5.0);
+  c1 = 2 / (std::pow(iterate.n_elem + 1.3, 2) + mu_eff);
+  alphacov = 2.0;
+  cmu = 2.0 * (mu_eff - 2.0 + 1.0 / mu_eff) / 
+      (std::pow(iterate.n_elem + 2.0, 2) + alphacov * mu_eff / 2);
+  cmu = std::min(1.0 - c1, cmu);
+
+  // Finalize the weights vector by computed paramters.
+  weights = weightPolicy.Generate(iterate.n_elem, c1, cmu);
+
+  // Controlling.
+  dsigma = 1 + csigma + 2 * 
+      std::max(std::sqrt((mu_eff-1)/(iterate.n_elem+1)) - 1, 0.0);
```

Comment:
```suggestion
      std::max(std::sqrt((mu_eff - 1) / (iterate.n_elem + 1)) - 1, 0.0);
```

Please use camel casing for all names.